src/fu-tool.c

```diff
@@ -1304,12 +1304,52 @@ fu_util_install_release(FuUtilPrivate *priv, FwupdRelease *rel, GError **error)
 }
 
 static gboolean
-fu_util_update_all(FuUtilPrivate *priv, GError **error)
+fu_util_update(FuUtilPrivate *priv, gchar **values, GError **error)
 {
 	g_autoptr(GPtrArray) devices = NULL;
 	gboolean no_updates_header = FALSE;
 	gboolean latest_header = FALSE;
 
+	if (priv->flags & FWUPD_INSTALL_FLAG_ALLOW_OLDER) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_ARGS,
+				    "--allow-older is not supported for this command");
+		return FALSE;
+	}
+
+	if (priv->flags & FWUPD_INSTALL_FLAG_ALLOW_REINSTALL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_ARGS,
+				    "--allow-reinstall is not supported for this command");
+		return FALSE;
+	}
+
+	if (!fu_util_start_engine(priv,
+				  FU_ENGINE_LOAD_FLAG_COLDPLUG | FU_ENGINE_LOAD_FLAG_HWINFO |
+				      FU_ENGINE_LOAD_FLAG_REMOTES,
+				  error))
+		return FALSE;
+
+	/* DEVICE-ID and GUID are acceptable args to update */
+	for (guint idx = 0; idx < g_strv_length(values); idx++) {
```

Comment:
Is values[0] == NULL for the case where you don't provide args?

Reply:
Yes, it's `0x0` in the instance of `fwupdtool update` with no args.